internal/transport/transport_test.go

```diff
@@ -1663,6 +1663,82 @@ func (s) TestReadGivesSameErrorAfterAnyErrorOccurs(t *testing.T) {
 	}
 }
 
+func (s) TestServerWithClientSendingWrongMethod(t *testing.T) {
+	server := setUpServerOnly(t, 0, &ServerConfig{}, suspended)
+	defer server.stop()
+	// Create a client directly to not couple what you can send to API of http2_client.go
+	mconn, err := net.Dial("tcp", server.lis.Addr().String())
+	if err != nil {
+		t.Fatalf("Client failed to dial: %v", err)
+	}
+	defer mconn.Close()
+
+	if n, err := mconn.Write(clientPreface); err != nil || n != len(clientPreface) {
+		t.Fatalf("mconn.Write(clientPreface) = %d, %v, want %d, <nil>", n, err, len(clientPreface))
+	}
+	// success chan indicates that reader received a RSTStream from server
+	success := make(chan struct{})
```

Comment:
`transport_test.go` doesn't use `testutils.Channel` because the former was written before the latter. It might be a good idea to use `testutils.Channel` here. Here is an [example](https://github.com/grpc/grpc-go/blob/master/balancer/rls/internal/client_test.go#L95-L117).

Reply:
Changed to use testutils.Channel.